packages/core/src/IconButton.js

```diff
@@ -20,16 +22,24 @@ const TransparentButton = styled(Button)`
   ${applyVariations('IconButton')}
 `
 
-const IconButton = ({ icon, ...props }) => (
-  <TransparentButton {...props}>
-    <div>{icon}</div>
-  </TransparentButton>
-)
+const IconButton = ({ icon, ...props }) => {
+  // TODO remove once migrated to SC4
+  const spreadProps = props
+  spreadProps[getSCMigrationRef()] = props.dsRef
+  delete spreadProps.dsRef
```

Comment:
Would this be nicer?
```
const spreadProps = {
  ...props,
  [getSCMigrationRef()] = props.dsRef,
}
```
And is there any reason to delete dsRef?

Reply:
It didn't seem like it, but I wanted to avoid confusing people by passing the same value as different props.